addons/binding/org.openhab.binding.opendaikin/src/main/java/org/openhab/binding/opendaikin/internal/discovery/OpenDaikinACUnitDiscoveryService.java

```diff
@@ -0,0 +1,149 @@
+        if (backgroundFuture != null && !backgroundFuture.isDone()) {
+            backgroundFuture.cancel(true);
+            backgroundFuture = null;
+        }
+
+        super.stopBackgroundDiscovery();
+    }
+
+    private Runnable createScanner() {
+        return () -> {
+            long timestampOfLastScan = getTimestampOfLastScan();
+            for (InetAddress broadcastAddress : getBroadcastAddresses()) {
+                logger.debug("Starting broadcast for {}", broadcastAddress.toString());
+
+                try (DatagramSocket socket = new DatagramSocket()) {
+                    socket.setBroadcast(true);
+                    socket.setReuseAddress(true);
+                    byte[] packetContents = UDP_PACKET_CONTENTS.getBytes(StandardCharsets.UTF_8);
+                    DatagramPacket packet = new DatagramPacket(packetContents, packetContents.length, broadcastAddress,
+                            REMOTE_UDP_PORT);
+
+                    // Send before listening in case the port isn't bound until here.
+                    socket.send(packet);
+
+                    while (receivePacketAndDiscover(socket)) {
+                    }
+                } catch (Exception e) {
+                    // Nothing to do here - the host couldn't be found, likely because it doesn't exist
+                }
+            }
+
+            removeOlderResults(timestampOfLastScan);
+        };
+    }
+
+    private boolean receivePacketAndDiscover(DatagramSocket socket) {
+        try {
+            // Use a one byte buffer since we don't really care about the contents.
+            byte[] buffer = new byte[1];
+            DatagramPacket incomingPacket = new DatagramPacket(buffer, buffer.length);
+            socket.setSoTimeout(1000 /* one second */);
+            socket.receive(incomingPacket);
+
+            String host = incomingPacket.getAddress().toString().substring(1);
+            logger.debug("Received packet from {}", host);
+            new OpenDaikinWebTargets(ClientBuilder.newClient(), host).getControlInfo();
+
+            ThingUID thingUID = new ThingUID(OpenDaikinBindingConstants.THING_TYPE_AC_UNIT, host.replace('.', '_'));
+            DiscoveryResult result = DiscoveryResultBuilder.create(thingUID)
+                    .withProperty(OpenDaikinConfiguration.HOST, host).withLabel("Daikin AC Unit (" + host + ")")
+                    .build();
+
+            logger.debug("Successfully discovered host {}", host);
+            thingDiscovered(result);
+        } catch (Exception e) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private List<InetAddress> getBroadcastAddresses() {
+        ArrayList<InetAddress> addresses = new ArrayList<>();
+
+        for (String broadcastAddress : NetUtil.getAllBroadcastAddresses()) {
+            try {
+                addresses.add(InetAddress.getByName(broadcastAddress));
+            } catch (UnknownHostException e) {
+            }
```

Comment:
Prefer to always log exceptions. You may never know it might be relevant in some use case. It can be logged to debug or trace if it's something that would trigger to much logging in normal usages.